browser-test/src/applicant/application_version_fast_forward.test.ts

```diff
@@ -0,0 +1,1000 @@
+        })
+
+        await test.step('edit program', async () => {
+          await civiformAdminActor.editProgram()
+        })
+
+        return await test.step('add questions to program and republish program', async () => {
+          await civiformAdminActor.addQuestionsToExistingBlocks([
+            {
+              block: Block.Fourth,
+              questions: [Question.H],
+              eligibilityValue: `${Question.H}-text-answer`,
+            },
+          ])
+
+          const programIdV3 =
+            civiformAdminActor.getProgramIdFromEditProgramUrl()
+          await civiformAdminActor.publish()
+          return programIdV3
+        })
+      })
+
+    // Verify we have a new program version id
+    expect(
+      programIdV2,
+      'Verify Program v2 and Program v3 have the different IDs.',
+    ).not.toEqual(programIdV3)
+    expect(
+      programIdV2,
+      'Verify Program v2 ID is less than Program v3 ID.',
+    ).toBeLessThan(programIdV3)
+
+    // Applicant submits application for program v1; it does not change to program v3
+    await test.step('As applicant - active program v3', async () => {
+      await test.step('As applicant - submit application', async () => {
+        await applicantActor.gotoNextPage()
+        await applicantActor.submitApplication()
+      })
+
+      await test.step('As applicant - check program list has one submitted application for program v1', async () => {
+        await applicantActor.gotoApplicantHomePage()
+        await expect(
+          applicantActor.getCardListLocator(ApplicationStatus.InProgress),
+        ).not.toBeAttached()
+        await expect(
+          applicantActor.getCardListLocator(ApplicationStatus.Submitted),
+        ).toBeAttached()
+
+        const headingLocator = applicantActor.getCardHeadingLocator(
+          ApplicationStatus.Submitted,
+        )
+        await expect(headingLocator).toBeAttached()
+
+        // Once submitted the program id on the edit button will be for the latest version
+        expect(
+          await applicantActor.getProgramIdFromLocator(headingLocator),
+        ).toBe(programIdV3)
+      })
+
+      await test.step('check program list shows eligible tag', async () => {
+        await expect(
+          applicantActor.getCardEligibleTagLocator(ApplicationStatus.Submitted),
+        ).toBeVisible()
+        await expect(
+          applicantActor.getCardNotEligibleTagLocator(
+            ApplicationStatus.Submitted,
+          ),
+        ).not.toBeAttached()
+      })
+    })
+
+    // Program Admin can view applications submitted for program v1
+    await test.step('As program admin - active program v3', async () => {
+      await test.step('Log in program admin', async () => {
+        // Log in Program Admin here and not at the start. If logged in before the program is created by
+        // a civiform admin, the program admin won't have permissions to the program.
```

Comment:
(huh that's an odd behavior)

Reply:
It's specifically a side effect of the fake users and how they are automatically given permissions to everything. Never been a problem before because we always have just been logging different actors in and out and in and out.

Reply:
Ahh good to know it's just a fake user thing